src/frontend/apps/markdown/MarkdownEditor/index.spec.tsx

```diff
@@ -0,0 +1,321 @@
+    };
+
+    return range;
+  };
+
+  afterEach(() => {
+    jest.resetAllMocks();
+    fetchMock.restore();
+  });
+
+  it('shows editor', async () => {
+    const markdownDocument = markdownDocumentMockFactory({
+      id: '1',
+      is_draft: true,
+      translations: [markdownTranslationMockFactory({ language_code: 'en' })],
+    });
+    const queryClient = new QueryClient();
+    const documentDeferred = new Deferred();
+    fetchMock.get('/api/markdown-documents/1/', documentDeferred.promise);
+
+    render(
+      wrapInIntlProvider(
+        <QueryClientProvider client={queryClient}>
+          <MarkdownEditor />
+        </QueryClientProvider>,
+      ),
+    );
+
+    await act(async () => documentDeferred.resolve(markdownDocument));
+
+    await screen.findByDisplayValue(markdownDocument.translations[0].title);
+    await waitFor(() =>
+      expect(
+        screen.getAllByText(markdownDocument.translations[0].content).length,
+      ).toEqual(2),
+    );
+
+    expect(screen.getByRole('button', { name: 'Save' })).not.toBeDisabled();
+    expect(screen.getByRole('button', { name: 'Publish' })).toBeDisabled();
+
+    fetchMock.patch(
+      `/api/markdown-documents/${markdownDocument.id}/save-translations/`,
+      markdownDocument,
+    );
+    act(() => {
+      userEvent.click(screen.getByRole('button', { name: 'Save' }));
+    });
+
+    await waitFor(() =>
+      expect(screen.getByRole('button', { name: 'Save' })).toBeDisabled(),
+    );
+    expect(screen.getByRole('button', { name: 'Publish' })).not.toBeDisabled();
+  });
+
+  it('change language', async () => {
+    window.scrollTo = jest.fn(); // required to test, see grommet
+    const markdownDocument = markdownDocumentMockFactory({
+      id: '1',
+      is_draft: true,
+      translations: [
+        markdownTranslationMockFactory({ language_code: 'en' }),
+        markdownTranslationMockFactory({ language_code: 'fr' }),
+      ],
+    });
+    const queryClient = new QueryClient();
+    fetchMock.get('/api/markdown-documents/1/', markdownDocument);
+
+    render(
+      wrapInIntlProvider(
+        <QueryClientProvider client={queryClient}>
+          <MarkdownEditor />
+        </QueryClientProvider>,
+      ),
+    );
+
+    // Wait for rendered content
+    await waitFor(() =>
+      expect(
+        screen.getAllByText(markdownDocument.translations[0].content).length, // en translation
+      ).toEqual(2),
+    );
+
+    // Force save... required to change the language
+    fetchMock.patch(
+      `/api/markdown-documents/${markdownDocument.id}/save-translations/`,
+      markdownDocument,
+    );
+    act(() => {
+      userEvent.click(screen.getByRole('button', { name: 'Save' }));
+    });
+    await waitFor(() =>
+      expect(screen.getByRole('button', { name: 'Save' })).toBeDisabled(),
+    );
+
+    // Change language to fr
+    userEvent.click(screen.getByRole('button', { name: 'Language: en' }));
+    await screen.findByTestId('drop_language_fr');
```

Comment:
Are you sure you have to await it ? In the `LanguageSelector` you don't do it. If yes store the return in a variable and use this variable in the next `act`.